include/rmm/logger.hpp

```diff
@@ -75,17 +75,22 @@ struct logger_wrapper {
 struct bytes {
   std::size_t value;
 
+  static constexpr char const* units(int index)
+  {
+    // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+    return std::array<char const*, 9>{"B", "KiB", "MiB", "GiB", "TiB", "PiB", "EiB", "ZiB", "YiB"}
```

Comment:
Wouldn't this construct a new array every time?

Reply:
No, it will just create a lookup table: https://godbolt.org/z/jcEvz4E5E

Reply:
@harrism was the usage of `at()` intentional here? You want the error checking? 

Also, with CTAD you can drop the template args, which is handy to avoid a mismatch in the size. 

```suggestion
    return std::array{"B", "KiB", "MiB", "GiB", "TiB", "PiB", "EiB", "ZiB", "YiB"}
```

Reply:
I used `at()` because clang-tidy complains about [].

Reply:
I think the original problem with the false non-coverage report for this line was due to line length (Which sometimes confuses gcov), not due to the `static`. CTAD helps shorten the line length, so I have a change that solves the problem with better code.